sdk-extensions/incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/fileconfig/OpenTelemetryConfigurationFactoryTest.java

```diff
@@ -0,0 +1,138 @@
+                        new OpenTelemetryConfiguration().withFileFormat("1"),
+                        spiHelper,
+                        closeables))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("Unsupported file format. Supported formats include: 0.1");
+    cleanup.addCloseables(closeables);
+  }
+
+  @Test
+  void create_Defaults() {
+    List<Closeable> closeables = new ArrayList<>();
+    OpenTelemetrySdk expectedSdk = OpenTelemetrySdk.builder().build();
+    cleanup.addCloseable(expectedSdk);
+
+    OpenTelemetrySdk sdk =
+        OpenTelemetryConfigurationFactory.getInstance()
+            .create(new OpenTelemetryConfiguration().withFileFormat("0.1"), spiHelper, closeables);
+    cleanup.addCloseable(sdk);
+    cleanup.addCloseables(closeables);
+
+    assertThat(sdk.toString()).isEqualTo(expectedSdk.toString());
+  }
+
+  @Test
+  void create_Configured() {
+    List<Closeable> closeables = new ArrayList<>();
+    OpenTelemetrySdk expectedSdk =
+        OpenTelemetrySdk.builder()
+            .setLoggerProvider(
+                SdkLoggerProvider.builder()
+                    .setLogLimits(
+                        () ->
+                            LogLimits.builder()
+                                .setMaxAttributeValueLength(1)
+                                .setMaxNumberOfAttributes(2)
+                                .build())
+                    .addLogRecordProcessor(
+                        io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor.builder(
+                                OtlpGrpcLogRecordExporter.getDefault())
+                            .build())
+                    .build())
+            .build();
+    cleanup.addCloseable(expectedSdk);
+
+    OpenTelemetrySdk sdk =
+        OpenTelemetryConfigurationFactory.getInstance()
+            .create(
+                new OpenTelemetryConfiguration()
+                    .withFileFormat("0.1")
+                    .withLoggerProvider(
+                        new LoggerProvider()
+                            .withLimits(
+                                new LogRecordLimits()
+                                    .withAttributeValueLengthLimit(1)
+                                    .withAttributeCountLimit(2))
+                            .withProcessors(
+                                Collections.singletonList(
+                                    new LogRecordProcessor()
+                                        .withBatch(
+                                            new BatchLogRecordProcessor()
+                                                .withExporter(
+                                                    new LogRecordExporter()
+                                                        .withOtlp(new Otlp())))))),
+                spiHelper,
+                closeables);
+    cleanup.addCloseable(sdk);
+    cleanup.addCloseables(closeables);
+
+    assertThat(sdk.toString()).isEqualTo(expectedSdk.toString());
```

Comment:
These are pretty cool tests. We create a configuration model, parse / interpret it, and compare the `toString()` result to an expected version we build up manually. Its a bit strange to use `toString` as a stand in for equality, but I think it makes sense in this case because these aren't data carrier objects.